boskos/client/client.go

```diff
@@ -60,23 +61,47 @@ type Client struct {
 	// http is the http.Client used to interact with the boskos REST API
 	http http.Client
 
-	owner string
-	url   string
-	lock  sync.Mutex
+	owner       string
+	url         string
+	username    string
+	getPassword func() []byte
+	lock        sync.Mutex
 
 	storage storage.PersistenceLayer
 }
 
+// for test only
+var Testing = false
+
 // NewClient creates a Boskos client for the specified URL and resource owner.
 //
 // Clients created with this function default to retrying failed connection
 // attempts three times with a ten second pause between each attempt.
-func NewClient(owner string, url string) *Client {
+func NewClient(owner string, urlString, username, password string) *Client {
+
+	var getPassword func() []byte
+	if password != "" {
+		u, err := url.Parse(urlString)
+		if err != nil {
+			logrus.WithError(err).Fatalf("Failed to parse the url '%s'", urlString)
+		}
+		if (! Testing) && u.Scheme != "https" {
```

Comment:
Ugly (production code only for testing) ... I know.
But the existing unit tests set up a `http` server and check the output of the `main` function.
I need to make the unit tests pass. Idea of improvement?

Reply:
Hoping that this is OK ... found similar tradeoff

https://github.com/kubernetes/test-infra/blob/08eaee10c47af9a237132bb975902e2b3ac7ecf9/boskos/cmd/cli/cli.go#L74-L76

Reply:
There is an HTTPS server in the `httptest` package you can use.

Reply:
I guess you meant https://golang.org/pkg/net/http/httptest/#NewTLSServer
The http client `ts.Client()` is form the server ... probably for a reason:

It uses a self-signed certificate and we have to `--insecure-skip-tls-verify=true` in the tests (because we do not want to use `ts.Client()`). But i think that is OK.